Assignment 2.Rmd

```diff
@@ -81,20 +89,90 @@ plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender)
 
 plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.width=EDGE$count)
 
-````
+```
 
 ## Part II
 
 In Part II your task is to [look up](http://igraph.org/r/) in the igraph documentation and create a graph that sizes the student vertices in terms of the number of disciplinary actions they have recieved, and the teachers in terms of the number of disciplinary actions they have given out.
 
+```{r}
+EDGESTD <- count(D2, stid)
+names(EDGESTD) <- c("id","count")
+EDGETCH<- count(D2,tid)
+names(EDGETCH) <- c("id","count")
+total<- bind_rows(EDGESTD,EDGETCH)
+VERTEX2<-full_join(VERTEX, total, by="id")
+g2 <- graph.data.frame(total, directed=TRUE, vertices=VERTEX2)
+plot(g2,layout=layout.fruchterman.reingold, vertex.color=VERTEX2$gender, vertex.size=VERTEX2$count+10)
+```
+
 
 
 ## Part III
 
 Now practice with data from our class. Please create a **person-network** with the data set hudk4050-classes.csv. To create this network you will need to create a person-class matrix using the tidyr functions and then create a person-person matrix using `t()`. You will then need to plot a matrix rather than a data frame using igraph.
 
+```{r}
+# install.packages("tidyr")
+library(tidyr)
+# install.packages("dplyr")
+library(dplyr)
+### select required data
+D3<- read.csv("hudk4050-classes.csv", header=TRUE)
+D3<-select(D3,Q8,Q9,Q1,Q3,Q4,Q5,Q6,Q7)
+### combine name
+D4<- D3 %>% unite("Name",Q8:Q9, sep=" ", remove=FALSE)
+D5<-select(D4,Name,Q1,Q3,Q4,Q5,Q6,Q7)
+D6<-drop(D5[3:55,])
+attach(D6)
+```
+
+```{r}
+### gather
+D7<- gather(D6,classnumber,value=class,2:7)
+### modify classes
+D7$class<-gsub("[a[[:space:]]","",D7$class)
+D7$class[67]<-"IFSF4090"
+D7$class[120]<-"EDPS4002"
+D7$class[34]<-"HUDK4050"
+D7$class[87]<-"HUDM4125"
+D7$class[140]<-"HUDM5026"
+D7$class[193]<-"HUDM5126"
```

Comment:
I appreciate your effort to clean the data in detail and very carefully. But if this is a larger dataset, perhaps we need other ways to deal with the messy data.